runtime/caml/memory.h

```diff
@@ -205,17 +205,19 @@ extern uintnat caml_use_huge_pages;
 #define DEBUG_clear(result, wosize)
 #endif
 
+#define Move_young_ptr(d, to) d->young_ptr = (value*)((uintnat)d->young_ptr to);
```

Comment:
I'm not a fan of this macro as it adds an indirection to understand what is going on. I think we should just do the arithmetic in words (as in upstream).
See suggestions below.